lib/ansible/module_utils/network/common/facts/facts.py

```diff
@@ -93,7 +48,9 @@ def get_network_resources_facts(self, net_res_choices, facts_resource_obj_map, r
             if not resource_facts_type:
                 resource_facts_type = self._gather_network_resources
 
-            restorun_subsets = self.gen_runable(resource_facts_type, frozenset(net_res_choices))
+            valid_subsets = net_res_choices
+            gather_subset = resource_facts_type
+            restorun_subsets = get_collector_names(valid_subsets, gather_subset=gather_subset)
```

Comment:
`net_res_choices` and `resource_facts_type` can be passed directly `get_collector_names` instead of seperately crearting those two varaible or instead `net_res_choices` can be renamed as `valid_subsets` and `resource_facts_type` to `gather_subset `